### PR TITLE
Contribute to tableQA

### PR DESCRIPTION
**Pull request by haotieu2001**
I created a table-question-answering folder and added about.md, data.ts file.

I added all content of data.ts

I can't preview it locally, I'm waiting for your feedback.

### CODE REVIEW

File: tasks/src/table-question-answering/data.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		},
+	],
+	models: 
+		[ 
+		{
+		       	description: "an empirically powerful pre-training approach to empower existing models with table reasoning skills.",
+		         id: "microsoft/tapex-base",
+		},
+		{
+			description: "TAPAS is a BERT-like transformers model pretrained on a large corpus of English data from Wikipedia in a self-supervised fashion.",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:46]: Can you simplify the description for a broader audience? If people want more technical information they can see the model card 🙂 Same applies for the other model.

File: tasks/src/table-question-answering/data.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		outputs: [
+			{ label: "Result",
+			  content: "7",
+			  type: "text",
+			},
+		],
+	},
+	metrics: 
+		[
+		{ description: "",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:32]: Can you add a description? Same with other one.

File: tasks/src/table-question-answering/data.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	datasets: [
+		{ 
+			description:"The Fewshot Table dataset consists of tables that naturally occur on the web, that are formatted as few-shot tasks for fine-tuning language models to improve their few-shot performance.",
+			id: "JeremyAlain/123_test",
+		},
+	],
+	demo: {
+		inputs: [
+			{ label: "Table",
+		          filename: "tableQA.jpg",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:13]: I think you can add a table component (you can see an example in tabular-classification task page)

File: tasks/src/table-question-answering/data.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import type { Task DataCustom } from "../Types";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+		{ 
+			description:"The Fewshot Table dataset consists of tables that naturally occur on the web, that are formatted as few-shot tasks for fine-tuning language models to improve their few-shot performance.",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:6]: Can we add a better documented canonical dataset, a couple including:
https://huggingface.co/datasets/wikisql
https://huggingface.co/datasets/msr_sqa
https://huggingface.co/datasets/wikitablequestions

**Review by merveenoyan — COMMENTED** [review]
Overall looks good to me, left a couple of suggestions🙂

File: tasks/src/table-question-answering/data.ts (hunk 1 above)
Comment (haotieu2001) [reply tasks/src/table-question-answering/data.ts:46]: Sure, I'm gonna change it.

File: tasks/src/table-question-answering/data.ts (hunk 2 above)
Comment (haotieu2001) [reply tasks/src/table-question-answering/data.ts:32]: I'm googling for definitions of these metrics, but I can't find resources.

Comment (haotieu2001) [reply tasks/src/table-question-answering/data.ts]: I've just changed the content. Please check it.
I see they use "denotation accuracy" in the TAPEX paper.

File: tasks/src/table-question-answering/about.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+## Use Cases
+
+### SQL execution
+
+You can use Table Question Answering model to simulate SQL execution by using provided table.
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/about.md:5]:
```suggestion
You can use the Table Question Answering models to simulate SQL execution by inputting a table.
```

Comment (merveenoyan) [reply tasks/src/table-question-answering/about.md:5]: Can you elaborate this a bit more, what do you input and what do you get after the inference?

File: tasks/src/table-question-answering/about.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+### Table Question Answering 
+
+Given the question and the table, the task is to answer the question based on the table, You can use Table Question Answering model to answer question based on the table.
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/about.md:9]:
```suggestion
Table Question Answering models are capable of answering questions based on a table.
```
It was a bit long and ambiguous.

File: tasks/src/table-question-answering/about.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+You can use Table Question Answering model to simulate SQL execution by using provided table.
+
+### Table Question Answering 
+
+Given the question and the table, the task is to answer the question based on the table, You can use Table Question Answering model to answer question based on the table. 
+
+## Inference 
+
+You can infer with TableQA models with the 🤗 Transformers library.
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/about.md:13]:
```suggestion
You can infer with TableQA models using the 🤗 Transformers library.
```

File: tasks/src/table-question-answering/about.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+model = AutoModelForSeq2SeqLM.from_pretrained("microsoft/tapex-large-finetuned-wtq")
+
+# prepare table + question
+data = {"Actors": ["Brad Pitt", "Leonardo Di Caprio", "George Clooney"], "Number of movies": ["87", "53", "69"]}
+table = pd.DataFrame.from_dict(data)
+question = "how many movies does Leonardo Di Caprio have?"
+
+encoding = tokenizer(table, question, return_tensors="pt")
+
+# let the model generate an answer autoregressively
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/about.md:29]: We try to avoid using terms to have a broader audience 🙂

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot for working on this, left suggestions regarding grammar & technical writing 🙂

**Comment by haotieu2001** [comment]
@merveenoyan sure.

**Comment by haotieu2001** [comment]
@merveenoyan I haven't known the template in the placeholder folder before :)

File: tasks/src/table-question-answering/data.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+			id: "JeremyAlain/123_test",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:7]: Can we put a maintained dataset as I see the latest issue was opened 3 months ago and hasn't yet resolved.

File: tasks/src/table-question-answering/data.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+			id: "wikisql",
+		},
+	],
+	demo: {
+		inputs: [
+			{ table: [
+				["Rank", "Name", "No.of reigns", "Combined days"],
+				[1, 'lou Thesz', 3, 3749],
+                                [2, 'Ric Flair', 8, 3103],
+                                [3, 'Harley Race', 7, 1799],
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:20]: I think six lines are too long, we can have three lines, WDYT?
Ekran Resmi 2022-10-14 12 29 04

File: tasks/src/table-question-answering/data.ts [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+                                [4, 'Dory Funk Jr.', 1, 1563],
+                                [5, 'Dan Severn', 2, 1559],
+                                [6, 'Gene Kiniski', 1, 1131]
+			],
+			  type: "tabular",
+			 },
+			 
+			 { label: "Question",
+		           content: "what is the number of reigns for Harley Race",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:29]:
```suggestion
		           content: "What is the number of reigns for the Harley Race?",
```

File: tasks/src/table-question-answering/data.ts [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+		outputs: [
+			{ label: "Result",
+			  content: "7",
+			  type: "text",
+			},
+		],
+	},
+	metrics: 
+		[
+		{ description: "checks whetherchecks whether the predicted answer(s) is equal to the ground-truth answer(s).",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:43]:
```suggestion
		{ description: "Checks whether the predicted answer(s) is equal to the ground-truth answer(s).",
```
Equal by means of what? Can you specify? Is it like BLEU score?

File: tasks/src/table-question-answering/data.ts [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+		  id: "denotation accuracy",
+		},
+	],
+	models: 
+		[ 
+		{
+		       	description: "You can use the raw model for simulating neural SQL execution, i.e., employ TAPEX to execute a SQL query on a given table.",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:50]:
```suggestion
		       	description: "A table question answering model that is capable of neural SQL execution, i.e., employ TAPEX to execute a SQL query on a given table.",
```

File: tasks/src/table-question-answering/data.ts [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+		         id: "microsoft/tapex-base",
+		},
+		{
+			description: "You can use this model for answering questions related to a table.",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:54]:
```suggestion
			description: "A robust table question answering model.",
```

File: tasks/src/table-question-answering/data.ts [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+			id: "google/tapas-base-finetuned-wtq",
+
+		},
+	],
+	summary: "Table Question Answering (Table QA) refers to providing precise answers from tables to answer a user's question.",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:59]:
```suggestion
	summary: "Table Question Answering (Table QA) is the answering a question about an information on a given table.",
```

**Review by merveenoyan — COMMENTED** [review]
Overall looks very good ❤️ Happy to merge after my comments are addressed 🙂 great job!

**Comment by merveenoyan** [comment]
Partially fixes #358

File: tasks/src/table-question-answering/data.ts (hunk 12 above)
Comment (haotieu2001) [reply tasks/src/table-question-answering/data.ts]: I'm googling a lot, but I can't find a clear definition for this metric.

File: tasks/src/table-question-answering/data.ts [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+		outputs: [
+			{ label: "Result",
+			  content: "7",
+			  type: "text",
+			},
+		],
+	},
+	metrics: 
+		[
+		{ description: "checks whether the predicted answer(s) is equal to the ground-truth answer(s).",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:40]:
```suggestion
		{ description: "Checks whether the predicted answer(s) is same as the ground-truth answer(s).",
```
small grammar nit 🙂

File: tasks/src/table-question-answering/data.ts [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+		  id: "denotation accuracy",
```
Comment (merveenoyan) [inline tasks/src/table-question-answering/data.ts:41]:
```suggestion
		  id: "Denotation Accuracy",
```
small nit

**Review by merveenoyan — APPROVED** [review]
I'm happy to merge, great work!

**Comment by haotieu2001** [comment]
@merveenoyan Thanks a lot, Do I need to close this PR?

**Comment by merveenoyan** [comment]
@haotieu2001 if you commit my suggestions I will merge your PR.

File: tasks/src/table-question-answering/data.ts (hunk 17 above)
Comment (haotieu2001) [reply tasks/src/table-question-answering/data.ts]: I've just fixed it.

**Comment by merveenoyan** [comment]
@haotieu2001 Can you replace _equal_ with _same_? _equal_ sounds a bit quantitative meanwhile you're trying to check if the words are the same.

**Comment by haotieu2001** [comment]
Done.

**Comment by merveenoyan** [comment]
@haotieu2001 congratulations! can I have your 🤗 Hub handle so I can add you at the end of the page?

**Comment by haotieu2001** [comment]
Thanks, this is my Hugingface account https://huggingface.co/haotieu

**Pull request merged**